src/evaluators/UnaryExpression.js

```diff
@@ -29,13 +31,124 @@ import {
+
+  // 4. If IsUnresolvableReference(ref) is true, then
+  if (Environment.IsUnresolvableReference(realm, ref)) {
+    // a. Assert: IsStrictReference(ref) is false.
+    invariant(!Environment.IsStrictReference(realm, ref), "did not expect a strict reference");
+
+    // b. Return true.
+    return realm.intrinsics.true;
+  }
+
+  // 5. If IsPropertyReference(ref) is true, then
+  if (Environment.IsPropertyReference(realm, ref)) {
+    // a. If IsSuperReference(ref) is true, throw a ReferenceError exception.
+    if (Environment.IsSuperReference(realm, ref)) {
+      throw realm.createErrorThrowCompletion(realm.intrinsics.ReferenceError);
+    }
+
+    // b. Let baseObj be ! ToObject(GetBase(ref)).
+    let base = Environment.GetBase(realm, ref);
+    // Constructing the reference checks that base is coercible to an object hence
+    invariant(base instanceof ConcreteValue || base instanceof AbstractObjectValue);
+    let baseObj = base instanceof ConcreteValue ? To.ToObject(realm, base) : base;
+
+    // c. Let deleteStatus be ? baseObj.[[Delete]](GetReferencedName(ref)).
+    let deleteStatus = baseObj.$Delete(Environment.GetReferencedName(realm, ref));
+
+    // d. If deleteStatus is false and IsStrictReference(ref) is true, throw a TypeError exception.
+    if (!deleteStatus && Environment.IsStrictReference(realm, ref)) {
+      throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError);
+    }
+
+    // e. Return deleteStatus.
+    return new BooleanValue(realm, deleteStatus);
+  }
+
+  // 6. Else ref is a Reference to an Environment Record binding,
+  // a. Let bindings be GetBase(ref).
+  let bindings = Environment.GetBase(realm, ref);
+  invariant(bindings instanceof EnvironmentRecord);
+
+  // b. Return ? bindings.DeleteBinding(GetReferencedName(ref)).
+  let referencedName = Environment.GetReferencedName(realm, ref);
+  invariant(typeof referencedName === "string");
+  return new BooleanValue(realm, bindings.DeleteBinding(referencedName));
+}
+
+function generateRuntimeCall(ref: Value | Reference, ast: BabelNodeUnaryExpression, strictCode: boolean, realm: Realm) {
+  let args = [];
+  let [thisArg, propName] = ref instanceof Reference ? [ref.base, ref.referencedName] : [];
+  if (thisArg instanceof Value) args = [thisArg];
+  if (propName !== undefined && typeof propName !== "string") args.push(propName);
+  if (thisArg instanceof Value) {
+    Leak.leakValue(realm, thisArg, ast.loc);
+  }
+  return AbstractValue.createTemporalFromBuildFunction(realm, Value, args, nodes => {
+    let arg;
+    if (thisArg instanceof Value) {
+      if (typeof propName === "string") {
+        arg = t.isValidIdentifier(propName)
+          ? t.memberExpression(nodes[0], t.identifier(propName), false)
+          : t.memberExpression(nodes[0], t.stringLiteral(propName), true);
+      } else {
+        arg = t.memberExpression(nodes[0], nodes[1], true);
+      }
+    } else {
+      arg = nodes[0];
```

Comment:
This line isn't covered by serializer tests. So it's not clear to me if it works and which case it covers.